boards/arm/actinius_icarus_som_dk/board_ns.c

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2022 Actinius
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/init.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/gpio.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(board_control, CONFIG_BOARD_ICARUS_SOM_DK_LOG_LEVEL);
+
+static int board_actinius_icarus_som_dk_init(const struct device *dev)
+{
+	const struct gpio_dt_spec sim =
+		GPIO_DT_SPEC_GET(DT_NODELABEL(sim_select), sim_gpios);
+
+	ARG_UNUSED(dev);
+
+	if (!device_is_ready(sim.port)) {
+		LOG_ERR("The SIM Select Pin port is not ready");
+
+		return -ENODEV;
+	}
+
+	if (DT_ENUM_IDX(DT_NODELABEL(sim_select), sim) == 0) {
+		(void)gpio_pin_configure_dt(&sim, GPIO_OUTPUT_HIGH);
+		LOG_INF("eSIM is selected");
+	} else {
+		(void)gpio_pin_configure_dt(&sim, GPIO_OUTPUT_LOW);
+		LOG_INF("External SIM is selected");
+	}
+
+	return 0;
```

Comment:
So this is a library that is shared by multiple boards by the same vendor, right?

I created `boards/common` when I was adding the runners infrastructure with the intention that more common board files could go here. Maybe `boards/common/actinius/` could hold a helper library that you could stash this in?

It's interesting because this is something that hardware model v2 (https://github.com/zephyrproject-rtos/zephyr/issues/51831) seems to support naturally, since we would have `boards/actinius/actinius_icarus_som_dk` alongside `boards/actinius/actinius_some_other_board`, and you could more easily add `boards/actinius/helper_lib`.

Reply:
Should I make a helper lib in `boards/common` or shall we wait to do this directly with hardware model v2 later?

Reply:
> Should I make a helper lib in `boards/common` or shall we wait to do this directly with hardware model v2 later?

If you want this merged anytime soon, I would go with a new lib in boards/common

Reply:
Alright, I added a helper library in boards/common and changed the som dk to use it. Please let me know if it looks good to you and I will then squash and add a commit for changing the other boards as well asap.
I have a couple of questions:
- Is there a way to avoid having to explicitly source the lib Kconfig in Kconfig.defconfig?
- The dts bindings stay with the board, right?
- CMake complains

```
CMake Warning at ../../../CMakeLists.txt:831 (message):
  No SOURCES given to Zephyr library: boards__arm__actinius_icarus_som_dk

  Excluding target from build.
```
Any suggestions for that?
Thank you.